nnsmith/materialize/tensorflow/dialect.py

```diff
@@ -96,66 +98,75 @@ def requires(self, input_shapes: List[AbsTensor]) -> List[z3.ExprRef]:
         return cons
 
 
-class NHWCConv2d(UnaryOpBase):
-    in_dtypes = [(DType.float16,), (DType.float32,), (DType.float64,)]
+class NHWCConv2dBase(BinaryOpBase):
+    in_dtypes = [
+        (DType.float16, DType.float16),
+        (DType.float32, DType.float32),
+        (DType.float64, DType.float64),
+    ]
```

Comment:
```suggestion
    in_dtypes = DTYPE_GEN_FLOATS
```

Reply:
`BinaryOpBase` has defined the `in_dtypes` and `out_dtypes` so we needn't redefine them here. Will remove them in the next commit.

Reply:
true. also i just noticed my prior suggestion is kinda wrong...